src/main/java/seedu/address/model/person/Phone.java

```diff
@@ -15,19 +18,47 @@
             "Phone numbers can only contain numbers, and should be at least 3 digits long";
     public static final String PHONE_VALIDATION_REGEX = "\\d{3,}";
     public final String value;
+    public final ArrayList<String> phonelist;
 
     /**
      * Validates given phone number.
      *
      * @throws IllegalValueException if given phone string is invalid.
      */
+
+    public Phone(List<String> phone) throws IllegalValueException {
+        requireNonNull(phone);
+        String phoneStr = "";
+        ArrayList<String> phones = new ArrayList<String>();
+        for (String s : phone) {
+            String trimmedPhone = s.trim();
+            if (!isValidPhone(trimmedPhone)) {
+                throw new IllegalValueException(MESSAGE_PHONE_CONSTRAINTS);
+            }
+            phones.add(s);
+            phoneStr += s + "\n";
+
+        }
+        this.phonelist = phones;
+        value = phoneStr;
+    }
+
     public Phone(String phone) throws IllegalValueException {
         requireNonNull(phone);
-        String trimmedPhone = phone.trim();
-        if (!isValidPhone(trimmedPhone)) {
-            throw new IllegalValueException(MESSAGE_PHONE_CONSTRAINTS);
+
+        String[] numbers = phone.split("\n");
+
+        ArrayList<String> phones = new ArrayList<String>();
+
+        for (int i = 0; i < numbers.length; i++) {
```

Comment:
You can use the same implementation as I did here so as to avoid index-for-loops which are more error-prone:
https://github.com/CS2103AUG2017-W10-B2/main/blob/1b592d12a28c72b3288570f44de4484e2b9f172a/src/main/java/seedu/address/logic/parser/ParserUtil.java#L58

Changes will look like this:
```
List<String> numbers = Arrays.asList(phone.split("\n"));
List<String> phones = new ArrayList<>();
// Use a for-each-for-loop instead
for (String p : numbers) {
    String trimmedPhone = p.trim();
    if (!isValidPhone(trimmedPhone)) {
        throw new IllegalValueException(MESSAGE_PHONE_CONSTRAINTS);
    }
    phones.add(p);
}
```